File: src/sdk-reference/go/1/getting-started/snippets/document.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"os"

	"github.com/kuzzleio/sdk-go/kuzzle"
	"github.com/kuzzleio/sdk-go/protocol/websocket"
)

func main() {
	// Create websocket protocol, replace "kuzzle" with
	// your Kuzzle hostname like "localhost"
	c := websocket.NewWebSocket("kuzzle", nil)
	// Instanciate a Kuzzle client
	kuzzle, _ := kuzzle.NewKuzzle(c, nil)

	// Connect to the server.
	if err := kuzzle.Connect(); err != nil {
		log.Fatal(err)
		os.Exit(1)
	}
	fmt.Println("Connected!")

	// Once connected, we can start to create document content...
	content := json.RawMessage(`
		{
			"name": "Sirkis",
			"birthday": "1959-06-22",
			"license": "B"
		}
	`)

	// ... then save our document in "yellow-taxi" collection.
	if _, err := kuzzle.Document.Create(
		"nyc-open-data",
		"yellow-taxi",
		"some-id",
		content,
		nil,
	); err != nil {
		log.Fatal(err)
		os.Exit(1)
	}
	fmt.Println("New document added to yellow-taxi collection!")

	// Finally, we can disconnect the SDK.
	kuzzle.Disconnect()
}

File: src/sdk-reference/cpp/1/getting-started/snippets/document.cpp
#include <iostream>
#include "kuzzle.hpp"
#include "websocket.hpp"

int main(int argc, char * argv[]) {
  // Instantiate a Kuzzle client with a WebSocket connection.
  // Replace "kuzzle" with your actual Kuzzle hostname (e.g. "localhost")
  kuzzleio::WebSocket *ws = new kuzzleio::WebSocket("kuzzle");
  kuzzleio::Kuzzle *kuzzle = new kuzzleio::Kuzzle(ws);

  try {
    // Connect to the server.
    kuzzle->connect();
    std::cout << "Connected!" << std::endl;
  }
  catch(kuzzleio::KuzzleException &e) {
      std::cerr << e.what() << std::endl;
      exit(1);
  }

  // New document content
  std::string document = R"(
    {
      "name": "Sirkis",
      "birthday": "1959-06-22",
      "license": "B"
    }
  )";

  try {
    // Store the document in the "yellow-taxi" collection, itself in the
    // "nyc-open-data" data index
    std::string response = kuzzle->document->create(
        "nyc-open-data",
        "yellow-taxi",
        "document-id",
        document);
    std::cout << "Document created successfully: " << response << std::endl;
  }
  catch(kuzzleio::KuzzleException &e) {
      std::cerr << e.what() << std::endl;
      exit(1);
  }

  // Disconnect and free allocated resources
  kuzzle->disconnect();

  delete kuzzle;
  delete ws;

  return 0;
}
